engine-standalone-tracing/src/sputnik.rs

```diff
@@ -51,8 +51,10 @@ impl evm_gasometer::tracing::EventListener for TransactionTraceBuilder {
             Event::RecordCost { cost, snapshot } => {
                 self.current.gas_cost = EthGas::new(cost);
                 if let Some(snapshot) = snapshot {
-                    self.current.gas =
-                        EthGas::new(snapshot.gas_limit - snapshot.used_gas - snapshot.memory_gas);
+                    if let gas = snapshot.gas_limit - snapshot.used_gas - snapshot.memory_gas >= 0 {
```

Comment:
No. It's wrong. The `gas` var is boolean here. Probably it should be written like this:
```rust
let gas = snapshot.gas_limit - snapshot.used_gas - snapshot.memory_gas;

if gas >= 0 {
    self.current.gas = EthGas::new(gas);
}
```

Reply:
Still getting clippy errors after the fix, says the boolean statement will always resolve to true.

Reply:
Yes. It's because type `u64` can't be negative 😄

Reply:
```rust
if let Some(gas) = snapshot.gas_limit.checked_sub(snapshot.used_gas)
    .and_then(|gas| gas.checked_sub(snapshot.memory_gas)) {
    self.current.gas = EthGas::new(gas);
}
```

Reply:
Yes, thank you, I have been able to get that far in the rust book :) Can I maybe skip storing `snapshot.gas_limit - snapshot.used_gas - snapshot.memory_gas;` in a variable, and instead evaluate if it is `>= 0` directly, or will it not make a difference?

Reply:
Oh, that's nice, thanks for the help @aleksuss!

Reply:
But the question is, should we skip assigning value in case of overflow, or should we assign `0` to rewrite the previous value of `self.current.gas`?
cc @birchmd